src/react-components/input/Slider.js

```diff
@@ -28,105 +24,56 @@ function updateGradient(el, min, max, value) {
+      value => {
+        setDisplayValue(value);
+        updateGradient(inputRef.current, min, max, value);
+      },
+      [inputRef, min, max, setDisplayValue]
+    );
 
-  const stopDragging = useCallback(
-    () => {
-      setIsDragging(false);
-    },
-    [setIsDragging]
-  );
-
-  const onUpdateValue = useCallback(
-    v => {
-      setDisplayValue(v);
-      if (onChange) {
-        onChange(v);
+    useImperativeHandle(ref, () => ({
+      setValue: value => {
+        updateValue(value);
       }
-    },
-    [onChange, setDisplayValue]
-  );
-
-  const drag = useCallback(
-    e => {
-      if (!isDragging) return;
-      const viewportOffset = myRoot.current.getBoundingClientRect();
-      const t = Math.max(0, Math.min((e.clientX - viewportOffset.left) / viewportOffset.width, 1));
-      const num = round(step, min + t * (max - min));
-      onUpdateValue(num);
-    },
-    [isDragging, step, min, max, onUpdateValue]
-  );
+    }));
 
-  useEffect(
-    () => {
-      window.addEventListener("mouseup", stopDragging);
-      window.addEventListener("mousemove", drag);
-
-      updateGradient(inputRef.current, min, max, currentValue);
-
-      return () => {
-        window.removeEventListener("mouseup", stopDragging);
-        window.removeEventListener("mousemove", drag);
-      };
-    },
-    [drag, stopDragging, inputRef, min, max, currentValue]
-  );
-
-  return (
-    <div className={classNames(styles.numberWithRange, className)}>
-      <div
-        ref={myRoot}
-        className={classNames(styles.rangeSlider)}
-        disabled={disabled}
-        onMouseDown={e => {
-          e.preventDefault();
-          if (disabled) return;
-          setIsDragging(true);
-          const viewportOffset = myRoot.current.getBoundingClientRect();
-          const t = Math.max(0, Math.min((e.clientX - viewportOffset.left) / viewportOffset.width, 1));
-          const num = round(step, min + t * (max - min));
-          const finalNum = parseFloat(num.toFixed(digits));
-          onUpdateValue(finalNum);
-        }}
-      >
-        <input
-          ref={inputRef}
-          type="range"
-          step={step}
-          min={min}
-          max={max}
-          value={currentValue}
-          disabled={disabled}
-          onChange={e => {
-            if (disabled) return;
-            const num = round(step, parseFloat(e.target.value));
-            const finalNum = parseFloat(num.toFixed(digits));
-            onUpdateValue(finalNum);
-          }}
-        />
+    return (
+      <div className={classNames(styles.numberWithRange, className)}>
+        <div className={classNames(styles.rangeSlider)} disabled={disabled}>
+          <input
+            ref={inputRef}
+            type="range"
+            step={step}
+            min={min}
+            max={max}
+            value={displayValue}
+            disabled={disabled}
+            onChange={e => {
+              if (disabled) return;
```

Comment:
Don't think its possible for `onChange` to be fired from a disabled `input`

Reply:
Removed